src/nomad_simulations/schema_packages/atoms_state.py

```diff
@@ -14,145 +13,301 @@
     from nomad.metainfo import Section
     from structlog.stdlib import BoundLogger
 
-from nomad_simulations.schema_packages.utils import RussellSaundersState, log
+from nomad_simulations.schema_packages.data_types import (
+    positive_float,
+    positive_int,
+    strictly_positive_int,
+    unit_float,
+)
+from nomad_simulations.schema_packages.utils import log
 
-# TODO rename this file particles_state.py or place ParticleState in model_system.py
-# TODO and make separate module files for AtomsState, CGBeadState, etc.
+# @JFRudzinski: TODO rename this file particles_state.py or place ParticleState in model_system.py
+# @JFRudzinski: TODO and make separate module files for AtomsState, CGBeadState, etc.
 
 
-class OrbitalsState(Entity):
-    """
-    A base section used to define the orbital state of an atom.
-    """
-
-    # TODO add check for `j_quantum_number` and `mj_quantum_number`
-    # TODO: add the relativistic kappa_quantum_number
-
+class BaseSpinOrbitalState(Entity):
     n_quantum_number = Quantity(
-        type=np.int32,
+        type=strictly_positive_int(),
         description="""
-        Principal quantum number of the orbital state.
+        Principal quantum number (n) of the electronic state. Must be > 0.
         """,
     )
 
-    l_quantum_number = Quantity(
-        type=np.int32,
-        description="""
-        Azimuthal quantum number of the orbital state. This quantity is equivalent to `l_quantum_symbol`.
-        """,
-    )
+    @property
+    def _name(self) -> str:
+        raise NotImplementedError('Subclasses must implement this method.')
 
-    l_quantum_symbol = Quantity(
-        type=str,
-        description="""
-        Azimuthal quantum symbol of the orbital state, "s", "p", "d", "f", etc. This
-        quantity is equivalent to `l_quantum_number`.
-        """,
-    )
+    @property
+    def _degeneracy(self) -> int:
```

Comment:
you normally use property decorator to cache the variable but it appears you did not implement it.